pygmt/tests/test_nearneighbor.py

```diff
@@ -0,0 +1,131 @@
+"""
+Tests for nearneighbor.
+"""
+import os
+
+import pytest
+import xarray as xr
+from pygmt import nearneighbor, which
+from pygmt.datasets import load_sample_bathymetry
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import data_kind
+
+TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+TEMP_GRID = os.path.join(TEST_DATA_DIR, "tmp_grid.nc")
+
+
+@pytest.fixture(scope="module", name="ship_data")
+def fixture_ship_data():
+    """
+    Load the data from the sample bathymetry dataset.
+    """
+    return load_sample_bathymetry()
+
+
+def test_nearneighbor_input_file():
+    """
+    Run nearneighbor by passing in a filename.
+    """
+    fname = which("@tut_ship.xyz", download="c")
+    output = nearneighbor(
+        data=fname, spacing="5m", region=[245, 255, 20, 30], search_radius="10m"
+    )
+    assert isinstance(output, xr.DataArray)
+    assert output.gmt.registration == 0  # Gridline registration
+    assert output.gmt.gtype == 0  # Cartesian type
```

Comment:
Add some more checks to ensure the output grid shape and statistics are ok.

```suggestion
    assert output.gmt.gtype == 0  # Cartesian type
    assert output.shape == (121, 121)
    npt.assert_allclose(output.mean(), -2378.2385)
```